android/build.gradle

```diff
@@ -207,3 +207,80 @@ dependencies {
 }
+
+def isGHExampleApp() {
+    return project.hasProperty('isGHExampleApp') && project.property('isGHExampleApp') == "true"
+}
+
+def getAbsoluteCodegenArtifactsPaperDestination() {
+    if (!project.hasProperty('codegenArtifactsPaperDestination')) {
+        throw new Exception('[react-native-gesture-handler] Please fill codegenArtifactsPaperDestination variable in android/gradle.properties to point to the correct path to generated specs for paper')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsPaperDestination')}"
+}
+
+def getAbsoluteCodegenArtifactsSource() {
+    if (!project.hasProperty('codegenArtifactsSource')) {
+        throw new Exception('[react-native-gesture-handler] Please fill codegenArtifactsSource variable in android/gradle.properties to point to the correct path to codegenerated artifacts')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsSource')}"
+}
+
+
+tasks.register("copyCodegenArtifacts") {
+    group 'After build tasks'
+    description 'Task which copies codegen artifacts to paper architecture'
+    
+    if (!isGHExampleApp() || !isNewArchitectureEnabled()) {
+        return
+    }
+
+    dependsOn tasks.generateCodegenArtifactsFromSchema
+
+    doLast {
+
+        def absoluteCodegenArtifactsPaperDestination = getAbsoluteCodegenArtifactsPaperDestination()
+        def absoluteCodegenArtifactsSource = getAbsoluteCodegenArtifactsSource()
+
+        def existingFiles = fileTree(absoluteCodegenArtifactsPaperDestination).matching {
+            include '**/*.java'
+        }
+
+        def generatedFiles = fileTree(absoluteCodegenArtifactsSource).matching {
+            include '**/*.java'
+        }
+
+        def existingFilesMap = [:]
+
+        existingFiles.forEach { existingFile ->
+            println existingFile
+            existingFilesMap[existingFile.name] = 1
+        }
+
+        generatedFiles.forEach { generatedFile -> 
+            if (!existingFilesMap.containsKey(generatedFile.name)) {
+                logger.warn("[react-native-gesture-handler] ${generatedFile.name} not found in paper dir, if it's used in Android you need to copy it manually and implement yourself before using auto-copy feature")
+            }
+        }
+
+        if (existingFiles.size() == 0) {
+            logger.warn("[react-native-gesture-handler] Paper destination with codegen interfaces is empty. This might be okay if you don't have any interfaces/delegates used in Android, if that's not the case please check if codegenArtifactsPaperDestination in android/gradle.properties is correct")
+        }
+
+        if (existingFiles.size() > generatedFiles.size()) {
+            throw new Exception("[react-native-gesture-handler] Number od generated artifacts should be greater than or equal to paper interfaces. Please check if codegenArtifactsSource in android/gradle.properties is correct")
```

Comment:
```suggestion
            logger.warn("[react-native-gesture-handler] Paper destination with codegen interfaces is empty. This might be okay if you don't have any interfaces/delegates used in Android, but if that's not the case, please check if path for `codegenArtifactsPaperDestination` property in android/gradle.properties configuration is correct.")
        }

        if (existingFiles.size() > generatedFiles.size()) {
            throw new Exception("[react-native-gesture-handler] Number od generated artifacts should be greater than or equal to paper interfaces. Please check if path for `codegenArtifactsSource` property in android/gradle.properties configuration is correct.")
```

Reply:
`please check if path for` shouldn't it be `please check if path in`? (The path is stored in that variable, not used by the variable)

Reply:
I actually changed this check as we discussed before.